shared/js/sync/engine.js

```diff
@@ -401,9 +409,19 @@ ccount`);
           return Promise.resolve();
         }
         return this._syncCollection(collectionName, collectionOptions)
-            .then(() => {
+            .then(syncResults2 => {
+          if (syncResults2.fetchToken) {
+            collectionOptions.fetchToken = syncResults2.fetchToken;
+            repeatNeeded = true;
+          } else {
+            repeatNeeded = false;
+          }
           this._haveUnsyncedConflicts[collectionName] = false;
         });
+      }).then(() => {
+        if (repeatNeeded) {
```

Comment:
Could we add a pref to enable/disable repetition, please? For the phone we don't want to fetch the whole history for now. At least until https://bugzilla.mozilla.org/show_bug.cgi?id=1231933 is fixed.